src/composables/nav.ts

```diff
@@ -21,3 +29,204 @@ export interface ActionItem {
   label: string
   show?: boolean | ((...args: any[]) => boolean)
 }
+
+export interface UseTabHistoryOpts {
+  /**
+   * The initial value of activeTab.
+   * The default value is tabs[0].value.
+   *
+   * When a valid tab value is found on window.location.search
+   * it will replace the initial value.
+   */
+  initial?: string
+
+  /**
+   * The param name to use on window.location.search.
+   * The default is "tab".
+   */
+  paramName?: string
+}
+
+/**
+ * useTabHistory
+ *
+ * Example Usage:
+ *
+ * const {activeTab, tabs} = useTabHistory([{label: "Tab One", value: "tab-1"}, {label: "Tab Two", value: "tab-2"}]})
+ *
+ * <Tabs v-model="activeTab" :tabs="tabs" />
+ *
+ * <div v-if="activeTab === 'tab-1'">Tab 1 Content</div>
+ * <div v-if="activeTab === 'tab-2'">Tab 2 Content</div>
+ *
+ * @param opts UseTabHistoryOpts
+ * @return {activeTab: Ref<string>, tabs: Ref<{labe: string, value: string}[]>}
+ */
+export const useTabHistory = (
+  initialTabs: MaybeRef<
+    {
+      label: string
+      value: string
+    }[]
+  >,
+  opts?: UseTabHistoryOpts
+) => {
+  const tabs = ref(initialTabs)
+  const config = {
+    initial: tabs?.value[0]?.value || "",
+    paramName: "tab",
```

Comment:
Since `opts.paramName` never overrides the config here, should it get passed at all? Or a different type for opts vs config? I expected `paramName: opts!.paramName ? opts!.paramName : 'tab'` here.

Reply:
Oh duh, my bad. My original question that I deleted was asking if `paramName` ever gets overridden. Why not just use tab and not make it configurable?

Reply:
It allows a page to potentially support two sets of tabbed navigation, as well as offering a bailout for some kind of unexpected naming conflict between another component that's mutating query params.  Both could be premature, supporting it just seemed trivial.

Reply:
I think nested tabs, is most likely case?  But potentially a bad UI as well!